examples/common/python/connectors/direct/tcs_listener/tcs_workorder_receipt_handler.py

```diff
@@ -35,311 +39,399 @@ def __init__(self, kv_helper):
+              as defined in EEA spec 7.2.3
+        Returns jrpc response as defined in 4.1
         """
 
+        wo_id = params["workOrderId"]
+        input_json_str = params["raw"]
         input_value = json.loads(input_json_str)
-        jrpc_id = input_value["id"]
 
-        # value retrieved is 'result' field as per Spec 7.2.5 Receipt Retrieve Response Payload
+        # Check if receipt for work order id is created or not
         value = self.kv_helper.get("wo-receipts", wo_id)
-        response['error'] = {}
 
         if value:
-            updater_value = input_value['params']
-            # WorkorderId already a part of receipt. And will be not change for a given receipt. Hence it's not stored in updater param.
-            del updater_value['workOrderId']
-
-            json_dict = json.loads(value)
-            updater_objects = json_dict['result']['updater']
-
-            if len(updater_objects) > 0:
-                # Updater Object is sorted based on index and then last index is chosen
-                index = int(sorted(updater_objects.keys())[-1]) + 1
+            # Receipt is created, validate update receipt request
+            status, err_msg = self.__validate_work_order_receipt_update_req(
+                    input_value
+                )
+            if status == True:
+                # Load previous updates to receipt
+                updated_receipt = None
+                updates_to_receipt = self.kv_helper.get("wo-receipt-updates", wo_id)
+                # If it is first update to receipt
+                if updates_to_receipt is None:
+                        updated_receipt = []
+                else:
+                    updated_receipt = json.loads(updates_to_receipt)
+                # Get last update to receipt
+                last_update = updated_receipt[len(updated_receipt)-1]
+                if last_update["updateType"] == ReceiptCreateStatus.COMPLETED.value:
+                    raise JSONRPCDispatchException(
+                        JRPCErrorCodes.INVALID_PARAMETER_FORMAT_OR_VALUE,
+                        "Receipt update status is set to completed, "
+                        "No further update allowed"
+                    )
+                # If last update to receipt is processed then below status
+                # are invalid
+                if last_update["updateType"] == ReceiptCreateStatus.PROCESSED.value:
+                    if input_value["params"]["updateType"] in [
+                        ReceiptCreateStatus.PENDING.value,
+                        ReceiptCreateStatus.FAILED.value,
+                        ReceiptCreateStatus.REJECTED.value
+                    ]:
+                        raise JSONRPCDispatchException(
+                            JRPCErrorCodes.INVALID_PARAMETER_FORMAT_OR_VALUE,
+                            "Current receipt status is set to processed, "
+                            "setting it to status pending, failed or rejected are not allowed"
```

Comment:
Consider replacing with -
"setting it to status "+input_value["params"]["updateType"]+" are not allowed"